selfservice/strategy/code/strategy_login.go

```diff
@@ -389,16 +391,21 @@ func (s *Strategy) PopulateLoginMethodSecondFactorRefresh(r *http.Request, f *lo
 	return s.PopulateMethod(r, f)
 }
 
-func (s *Strategy) PopulateLoginMethodIdentifierFirstCredentials(r *http.Request, f *login.Flow, _ ...login.FormHydratorModifier) error {
+func (s *Strategy) PopulateLoginMethodIdentifierFirstCredentials(r *http.Request, f *login.Flow, opts ...login.FormHydratorModifier) error {
 	if !s.deps.Config().SelfServiceCodeStrategy(r.Context()).PasswordlessEnabled {
 		// We only return this if passwordless is disabled, because if it is enabled we can always sign in using this method.
-		return idfirst.ErrNoCredentialsFound
+		return errors.WithStack(idfirst.ErrNoCredentialsFound)
+	}
+	o := login.NewFormHydratorOptions(opts)
+
+	// If the identity hint is nil and account enumeration mitigation is disabled, we return an error.
+	if o.IdentityHint == nil && !s.deps.Config().SecurityAccountEnumerationMitigate(r.Context()) {
+		return errors.WithStack(idfirst.ErrNoCredentialsFound)
```

Comment:
The problem here was, that if the user does not exist, we still trigger the "second step" which shows the "Send code" button, but after clicking that you get an error message that the account does not exist. 

This change "short-circuits" that, and immediately shows the "account doesn't exist" error after entering the email address (if account enumeration shouldn't be mitigated). 
If enumeration should be mitigated, this works the same way, the password strategy works.